great_expectations/expectations/regex_based_column_map_expectation.py

```diff
@@ -66,14 +84,35 @@ def _spark(cls, column, **kwargs):
         return column.rlike(cls.regex)
 
 
+@public_api
 class RegexBasedColumnMapExpectation(ColumnMapExpectation, ABC):
+    """Base class for RegexBasedColumnMapExpectations.
+
+    RegexBasedColumnMapExpectations facilitate regex parsing as the core logic for a Map Expectation.
```

Comment:
Do you think it would help to have a short usage example here as a code snippet?